src/Widgets/SecretItemRow.vala

```diff
@@ -94,22 +94,15 @@ public class Wallet.SecretItemRow : Gtk.ListBoxRow {
             revealer.reveal_child = false;
 
             GLib.Timeout.add (revealer.transition_duration, () => {
-                delete_secret ();
+                var schema = new Secret.Schema (
+                    "io.elementary.switchboard.wallet", Secret.SchemaFlags.NONE,
+                    "brand", Secret.SchemaAttributeType.STRING,
+                    "exp", Secret.SchemaAttributeType.STRING
+                );
+
+                Secret.password_clearv.begin (schema, secret_item.get_attributes (), null);
```

Comment:
Deletes all cards with the same brand and expiry date (?) Can there be more than one?